spark/src/main/scala/io/delta/sharing/spark/DeltaSharingSource.scala

```diff
@@ -579,13 +679,20 @@ case class DeltaSharingSource(
     val params = new RemoteDeltaFileIndexParams(
       spark, initSnapshot, deltaLog.client.getProfileProvider)
     val fileIndex = new RemoteDeltaBatchFileIndex(params, addFilesList)
-    CachedTableManager.INSTANCE.register(
-      params.path.toString,
-      idToUrl,
-      Seq(new WeakReference(fileIndex)),
-      params.profileProvider,
-      latestRefreshFunc
-    )
+    synchronized{
+      CachedTableManager.INSTANCE.register(
+        params.path.toString,
+        idToUrl,
+        Seq(new WeakReference(fileIndex)),
+        params.profileProvider,
+        latestRefreshFunc,
+        if (minUrlExpirationTimestamp.isDefined) {
+          minUrlExpirationTimestamp.get
+        } else {
+          lastQueryTableTimestamp + CachedTableManager.INSTANCE.preSignedUrlExpirationMs
+        }
+      )
+    }
```

Comment:
This synchronize is not needed if we define getMinUrlExpirationTimestamp (and sync within that).

Reply:
makes sense, updated.